zebra/zebra_vxlan.c

```diff
@@ -5271,21 +5271,6 @@ int zebra_vxlan_process_vrf_vni_cmd(struct zebra_vrf *zvrf, vni_t vni,
 		/* Remove L2VNI if present */
 		zebra_vxlan_handle_vni_transition(zvrf, vni, add);
 
-		/* check if the vni is already present under zvrf */
-		if (zvrf->l3vni) {
-			snprintf(err, err_str_sz,
-				 "VNI is already configured under the vrf");
-			return -1;
-		}
-
-		/* check if this VNI is already present in the system */
-		zl3vni = zl3vni_lookup(vni);
-		if (zl3vni) {
-			snprintf(err, err_str_sz,
-				 "VNI is already configured as L3-VNI");
-			return -1;
-		}
-
```

Comment:
SInce there is currently no locking between the validation stage and the apply stage isn't it possible for these vlaues to become invalid in between (race)? The NB infra cannot deal with the -1 return and will keep the config like it applied so that is not good either.. Perhaps it would be better to assert than return here, but keep the checks. In the mgmtd code the validation->apply stage should be protected by an internal lock in which case these asserts will never fail, b/c the race will never exist, but the code isn't there yet.

Reply:
Okay, thanks.
Anyway I will keep them for safe. And `-1` is not returned to NB, so it is ok.

Reply:
The only place this function is called is from the NB, so it is returning -1 to the NB. In both cases this is during the APPLY stage which doesn't allow for failure, so the effect is that it aborts the NB operation but keeps the config. I think an assert is better here, b/c otherwise the system is left in an inconsistent state. Would be nice to change it to assert if you can find the time.